File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    word_relay = [words[0]]

    for idx in range(1, len(words)):
        prev_word = word_relay[-1]
        curr_word = words[idx]

        if curr_word in word_relay or prev_word[-1] != curr_word[0] or len(curr_word) == 1:
            return [(idx % n) + 1, (idx // n) + 1]

        word_relay.append(curr_word)
    return [0, 0]

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
    let wordRelay = [words[0]];
    
    for(let i = 1 ; i < words.length ; i++){
        const prevWord = wordRelay[wordRelay.length - 1];
        const currWord = words[i];
        
        if(wordRelay.includes(currWord) || prevWord[prevWord.length - 1] !== currWord[0] || currWord.length === 1){
            return [(i % n) + 1, Math.floor(i / n) + 1];
        }
        wordRelay.push(currWord);
    }
    return [0, 0];
}